velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -251,7 +251,18 @@ RowVectorPtr NestedLoopJoinProbe::getOutput() {
     // Generate actual join output by processing probe and build matches, and
     // probe mismaches (for left joins).
     output = generateOutput();
+
+    // Check if driver should yield to prevent getting stuck in long processing
+    if (FOLLY_UNLIKELY(shouldYield())) {
+      if (output == nullptr) {
+        common::testutil::TestValue::adjust(
```

Comment:
Is there a way to not have this hook added here? You can add a sleep in Driver::runInternal::getOutput() hook. The hook execution is on-thread so it will be counted towards the driver slice time. Let's chat offline